addons/sale/models/sale_order_line.py

```diff
@@ -1155,7 +1155,7 @@ def _prepare_invoice_line(self, **optional_values):
             'is_downpayment': self.is_downpayment,
         }
         self._set_analytic_distribution(res, **optional_values)
-        if self.is_downpayment:
+        if self.is_downpayment and len(self.invoice_lines.filtered('is_downpayment')) > 0:
```

Comment:
this could be written

```suggestion
        if self.is_downpayment and self.invoice_lines.filtered('is_downpayment'):
```
or to not repeat, you could even put it in variable, eg:

```suggestion
        downpayment_lines = self.is_downpayment and self.invoice_lines.filtered('is_downpayment')
        if downpayment_lines:
            res['account_id'] = downpayment_lines.account_id[:1].id
```

Reply:
Done